classes/classes.py

```diff
@@ -0,0 +1,154 @@
+# Task 1
+
+class Snail:
+    def __init__(self, name = 'noname', species = 'unknown', age = 0, size = 0, status = 'inside'):
+        self.name = name
+        self.species = species
+        self.age = age
+        self.size = size
+        self.status = status
+
+    def age_group(self):
+        if self.age <= 15:
+            print('young')
+        elif self.age <= 60:
+            print('adult')
+        else:
+            print('old')
+
+    def size_group(self):
+        if self.size <= 20:
+            print('small')
+        elif self.size < 80:
+            print('medium')
+        else:
+            print('large')
+
+    def growth(self):
+        self.size += 1
+
+    def greet(self):
+        if self.status == 'inside':
+            print("Well, well. Look who's inside again.")
+        elif self.status == 'outside':
+            print("Hello, world!", " I am", self.name)
+        else:
+            print("I'm in a frightening, liminal space between states of being")
+
+# Task 2
+
+class RNA:
+    
+    def __init__(self, sequence):
+        self.sequence = str(sequence).upper()
+        
+    def translation(self):
+        seq = self.sequence
+        code = {'UUU' : 'F', 'UUC' : 'F', 'UUA' : 'L', 'UUG' : 'L', 'UCU' : 'S', 'UCC' : 'S', 'UCA' : 'S', 'UCG' : 'S', 'UAU' : 'Y', 'UAC' : 'Y', 'UAA' : 'stop', 'UAG' : 'stop', 'UGU' : 'C', 'UGC' : 'C', 'UGA' : 'stop', 'UGG' : 'W', 'CUU' : 'L', 'CUC' : 'L', 'CUA' : 'L', 'CUG' : 'L', 'CCU' : 'P', 'CCC' : 'P', 'CCA' : 'P', 'CCG' : 'P', 'CAU' : 'H', 'CAC' : 'H', 'CAA' : 'Q', 'CAG' : 'Q', 'CGU' : 'R', 'CGC' : 'R', 'CGA' : 'R', 'CGG' : 'R', 'AUU' : 'I', 'AUC' : 'I', 'AUA' : 'I', 'AUG' : 'M', 'ACU' : 'T', 'ACC' : 'T', 'ACA' : 'T', 'ACG' : 'T', 'AAU' : 'N', 'AAC' : 'N', 'AAA' : 'K', 'AAG' : 'K', 'AGU' : 'S', 'AGC' : 'S', 'AGA' : 'R', 'AGG' : 'R', 'GUU' : 'V', 'GUC' : 'V', 'GUA' : 'V', 'GUG' : 'V', 'GCU' : 'A', 'GCC' : 'A', 'GCA' : 'A', 'GCG' : 'A', 'GAU' : 'D', 'GAC' : 'D', 'GAA' : 'E', 'GAG' : 'E', 'GGU' : 'G', 'GGC' : 'G', 'GGA' : 'G', 'GGG' : 'G'}
+        protein =''
+        while seq[0:3] != 'AUG':
+            seq = seq[1:]
+        while len(seq) > 2:
+            if code[seq[0:3]] == 'stop':
+                break
+            protein += code[seq[0:3]]
+            seq = seq[3:]
+        print(protein)
+        
+    def reverse_transcription(self):
+        pairs = {'A':'T', 'U':'A', 'G':'C', 'C':'G'}
+        dna = ''
+        for base in self.sequence:
+            dna += pairs[base]
+        print(dna)
+
+# Task 3
+
+class PositiveSet(set):
+    def __init__(self, *args):
+        self.set = set()
+        for arg in args:
+            if arg > 0:
+                self.set.add(arg)
+    def add(self, *args):
+        for arg in args:
+            if arg > 0:
+                self.set.add(arg)
```

Comment:
Вариант рабочий, но здесь совсем не используются фишки наследования от `set`, например автоматический перенос в наш класс всех остальныъ операций с множествами. 
![image](https://user-images.githubusercontent.com/48160953/156925176-65587b05-ef70-4212-8333-a1aa5ad7453e.png)
Если отсюда убрать наследование, то ничего не изменится. К тому же обращаться ко множеству через атрибут `set` тоже не очень удобно. 
```suggestion
class PositiveSet(set):
    def __init__(self, *args):
        set_ = [arg for arg in args if arg > 0]
        super().__init__(set_)

    def add(self, *args):
        for arg in args:
            if arg > 0:
                super().add(arg)
```